lib/danger/environment_manager.rb

```diff
@@ -32,5 +32,26 @@ def fill_environment_vars
 
       self.scm = GitRepo.new # For now
     end
+
+    def ensure_danger_branches_are_setup
+      # As this currently just works with GitHub, we can use a github specific feature here:
+      pull_id = ci_source.pull_request_id
+      test_branch = request_source.base_commit
+
+      # Next, we want to ensure that we have a version of the current branch that at a know location
+      scm.exec "branch #{danger_base_branch} #{test_branch}"
+
+      # OK, so we want to ensure that we have a known head branch, this will always represent
+      # the head ( e.g. the most recent commit that will be merged. )
+      scm.exec "fetch origin +refs/pull/#{pull_id}/merge:#{danger_head_branch}"
```

Comment:
So we create local branches to make sure we have the right commits? Should we delete this branch after we're done?